third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -55,6 +54,33 @@ pub(crate) enum ValueImpl {
 
     ContainerRef(ContainerRef),
     IndexedRef(IndexedRef),
+
+    /// Delayed values are values that live outside of MoveVM and are processed in
+    /// a delayed (some may it call lazy) fashion, e.g., aggregators or snapshots.
+    /// The implementation stores a unique identifier so that the value can be
+    /// fetched and processed by native functions.
+    ///
+    /// Delayed values are sized, and the variant carries the information about
+    /// the serialized size of the external Move value. This allows to make sure
+    /// size information is known, e.g. for gas metering purposes.
+    ///
+    /// Delayed values should not be displayed in any way, to ensure we do not
+    /// accidentally introduce non-determinism if identifiers are generated at
+    /// random. For that reason, `Debug` is not derived for `ValueImpl` enum and
+    /// is implemented directly.
+    ///
+    /// Semantics:
+    ///   - Delayed values cannot be compared. An equality check results in a
+    ///     runtime error. As a result, equality for any Move value that contains
```

Comment:
Just to double check, since aggregator v2 is not on testnet yet, so we don't need to feature-gate this, right?

Reply:
Indeed! We already have feature flags for aggregators not enabled there, so once we do enabled them these APIs will start working